### PR TITLE
Increase material button minimum tap target height to 48dp

### LINKED ISSUE #615: Flat buttons are supposed to have a 48px high tap target
https://www.google.com/design/spec/components/buttons.html#buttons-usage

Right now, we don't enforce the 6px top and bottom margins.
Also, we don't tap target in that region like the spec says we should.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Increases the minimum tap target height for all material buttons to 48dp using an additional set of outer constraints. 

Also adjusts the height of the TimePicker by 12dp and the minimum size of a Snackbar with an action to 48dp.

Fixes flutter/flutter#615

### CODE REVIEW

File: packages/flutter/lib/src/material/flat_button.dart [hunk 1]
```diff
@@ -290,6 +290,7 @@ class FlatButton extends StatelessWidget {
       splashColor: _getSplashColor(theme, buttonTheme),
       elevation: 0.0,
       highlightElevation: 0.0,
+      outerPadding: const EdgeInsets.only(top: 6.0, bottom: 6.0),
```
Comment (goderbauer) [inline packages/flutter/lib/src/material/flat_button.dart]: Should this only be added if the button itself is sized too small or always?

**Review by goderbauer — COMMENTED** [review]
Can you add a test?

File: packages/flutter/lib/src/material/flat_button.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter/lib/src/material/flat_button.dart]: Good point - the spec isn't very clear but only claims a minimum touch target size of 48x48.  I updated outer padding to constraints so we can express this

**Comment by jonahwilliams** [comment]
Updating the FlatButton broke some tests for both the snackbar and the time picker.

In the time pickers case, the extra space consumed by the buttons shrinks the clock on the time picker.  Not sure whether the best approach would be to increase the overall size so that the clock doesn't change shapes or just update all of the tests

**Comment by Hixie** [comment]
Anything where this changed the layout you probably want to adjust so that the resulting layout is left unchanged.

If this changes the layout anywhere, please make sure to follow the breaking change protocol and also make sure to do a google3 roll immediately after landing this to handle any google3 breakage.

**Comment by jonahwilliams** [comment]
Since this is a major breaking change I will follow up in a week or so

**Comment by jonahwilliams** [comment]
There are two problematic tests which I need to do more investigation into before submitting

**Comment by jonahwilliams** [comment]
Something which might make this change easier is to expose the outer padding to each material button class to allow setting it to null, removing the padding.  We could default to null for a release so people could have time to fix their code the switch as another breaking change.  Similar issues for chips and other tap targets.

**Comment by Hixie** [comment]
I think we're better off just making the breaking change and moving on, having multiple staged rollouts is confusing and logistically challenging.

**Comment by jonahwilliams** [comment]
Closing in favor of https://github.com/flutter/flutter/pull/18369/

**Pull request closed without merging**